src/psyclone/tests/utilities.py

```diff
@@ -303,12 +303,17 @@ def _code_compiles(self, psy_ast, dependencies=None):
         :rtype: bool
 
         '''
-        kernel_modules = set()
+        modules = set()
         # Get the names of the modules associated with the kernels.
         # By definition, built-ins do not have associated Fortran modules.
         for invoke in psy_ast.invokes.invoke_list:
+            for symbol in invoke.schedule.symbol_table.containersymbols:
+                modules.add(symbol.name)
+            # For some PSyIR that container symbols may not be registered
```

Comment:
Suggest: "For some PSyIR (e.g. from xxx), there may be container symbols that have not been registered with the symboltable. Therefore we make sure that we at least insert symbols for the
modules associated with the CodedKernels."

Reply:
My initial comment was confusing, in fact the missing modules are from where the PSyIR doesn't reach yet. I updated the comment.

Also, I end up making the search for PSyIR modules more exhaustive, which in turn make me notice that not all dependencies need to be found and compiled since some are already pre-compiled. I ended up updating a few things in this utility function. I hope is for the better.

Reply:
I think it is better - thanks. Skipping files that aren't found made me a bit twitchy but I don't think we maintain a list of infrastructure modules anywhere as we rely on the appropriate Makefile to build the infrastructure libs. We could log/print a warning when this happens but that can be postponed to #1199.